src/nccl_ofi_rdma.cpp

```diff
@@ -1898,22 +1894,16 @@ static int receive_progress(nccl_net_ofi_rdma_req_t *req, bool add_to_pending)
 static int process_pending_reqs(nccl_net_ofi_rdma_ep_t *ep)
 {
 	int rc = 0;
-	nccl_ofi_deque_elem_t *deque_elem;
-	nccl_ofi_deque_t *pending_reqs_queue = ep->pending_reqs_queue;
 
 	while (true) {
-		rc = nccl_ofi_deque_remove_front(pending_reqs_queue, &deque_elem);
-		if (OFI_UNLIKELY(rc != 0)) {
-			NCCL_OFI_WARN("Failed to nccl_ofi_deque_remove_front: %d", rc);
-			return rc;
-		}
-
-		if (deque_elem == NULL) {
-			/* Deque is empty */
+		nccl_net_ofi_mutex_lock(&ep->pending_reqs_lock);
+		if (ep->pending_reqs_queue->empty()) {
```

Comment:
This code makes me nervous.  I'd rewrite as

```
nccl_net_ofi_rdma_req_t *req = NULL;
nccl_net_ofi_mutex_lock(&ep->pending_reqs_lock);
if (!ep->pending_reqs_queue->empty()) {
    req = ep->pending_reqs_queue->front();
    ep->pending_reqs_queue->pop_front();
}
nccl_net_ofi_mutex_unlock(&ep->pending_reqs_lock);
if (req == NULL) { break; }
```

Goal should be to minimize the number of conditional locking behaviors.

Reply:
Fixed in latest force push!